conans/test/functional/toolchains/env/test_complete.py

```diff
@@ -122,18 +122,22 @@ class Pkg(ConanFile):
             default_options = {"myopenssl:shared": True}
             exports_sources = "CMakeLists.txt", "main.cpp"
             generators = "CMakeDeps", "CMakeToolchain", "VirtualBuildEnv", "VirtualRunEnv"
+            apply_env = False
 
             def build(self):
                 cmake = CMake(self)
                 cmake.configure()
                 cmake.build()
                 self.run("mycmake")
                 self.output.info("RUNNING MYAPP")
+                filename = "conanrunenv-{}-{}".format(self.settings.build_type,
```

Comment:
Not a good UI,  Should we enable all the `env` collaborators a common interface to get the name? something like `VirtualRunEnv(self).filename`,  `VCVars(self).filename`, ... 
The bad part is that you need to import the class.

Reply:
But I have made it configurable, in the ``generate()`` method you can change that (same pattern as CMakeDeps)
So you cannot create a new instance, we would need to serialize somewhere the name of the file, and that would be ugly too...

Reply:
I have rewritten the whole ``env_scripts`` approach, now can be used for build, run (and potentially other environments at the user control), so this solves this UX issue. It was necessary some breaking changes, please review.